docs/guides/docs/intro-to-sway/typescript-sdk.mdx

```diff
@@ -0,0 +1,697 @@
+---
+title: Typescript Frontend
+category: Intro to Sway
+parent:
+  label: Guides
+  link: /guides
+---
+
+## Building the Frontend
+
+### Setup
+
+Initialize a new React app with TypeScript in the same parent folder as your contract using the command below.
+
+<TestAction
+id="create-frontend"
+action={{
+  name: 'runCommand',
+  commandFolder: 'guides-testing/fuel-project'
+}}
+/>
+
+```sh
+npx create-react-app frontend --template typescript
+```
+
+Let's go into the frontend folder:
+
+```sh
+cd frontend
+```
+
+Next, install the fuels Typescript and wallet SDKs in the frontend folder:
+
+<TestAction
+id="frontend-npm-install"
+action={{
+  name: 'runCommand',
+  commandFolder: 'guides-testing/fuel-project/frontend'
+}}
+/>
+
+```sh
+npm install fuels@0.66.1 @fuel-wallet/sdk@0.13.7 @fuel-wallet/react@0.13.7 --save
+```
+
+Generate types from your contract with `fuels typegen` by running:
+
+<TestAction
+id="frontend-gen-types"
+action={{
+  name: 'runCommand',
+  commandFolder: 'guides-testing/fuel-project/frontend'
+}}
+/>
+
+```sh
+npx fuels typegen -i ../contract/out/debug/*-abi.json -o ./src/contracts
+```
+
+Open the `src/App.tsx` file, and replace the boilerplate code with the template below:
+
+<TestAction
+id="fe-app-template"
+action={{
+  name: 'writeToFile',
+  filepath: 'guides-testing/fuel-project/frontend/src/App.tsx'
+}}
+/>
+
+<CodeImport
+  file="../../examples/intro-to-sway/frontend/src/App.tsx"
+  comment="fe_app_all"
+  commentType="//"
+  lang="tsx"
+/>
+
+Copy and paste the CSS code below in your `App.css` file to add some simple styling.
+
+<TestAction
+id="fe-css-template"
+action={{
+  name: 'writeToFile',
+  filepath: 'guides-testing/fuel-project/frontend/src/App.css'
+}}
+/>
+
+<CodeImport
+  file="../../examples/intro-to-sway/frontend/src/App.css"
+  comment="fe_css"
+  commentType="//"
+  lang="tsx"
+/>
+
+### Connecting to the contract
+
+Wrap your components with `FuelProvider` from `@fuel-wallet/react` to enable Fuel's custom React hooks for wallet functionalities in `index.tsx`.
+
+<TestAction
+id="fe-index-all"
+action={{
+  name: 'writeToFile',
+  filepath: 'guides-testing/fuel-project/frontend/src/index.tsx'
+}}
+/>
+
+<CodeImport
+  file="../../examples/intro-to-sway/frontend/src/index.tsx"
+  comment="fe_index_all"
+  commentType="//"
+  lang="tsx"
+/>
+
+At the top of the file, add your contract ID that you had earlier as a constant.
```

Comment:
This is out of place after editing the `index.tsx` file. I think it makes sense to edit `index.tsx` and then `App.tsx`.